src/providers/azure/mod.rs

```diff
@@ -418,12 +418,28 @@ impl Azure {
             .chain_err(|| "failed to get hostname")?;
         Ok(name)
     }
+
+    fn fetch_sku (&self) -> Result<Option<String>> {
+        const SKU_URL: &str = "metadata/instance/compute/sku?api-version=2017-08-01&format=text";
```

Comment:
For your task, I think you want to use the `vmSize` instead of the `sku` here.

Reply:
Ah, yup I think you'are right, for future reference:
 - [vmSize](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes)
 - [sku](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/cli-ps-findimage#table-of-commonly-used-windows-images)